relayer/processor/path_processor.go

```diff
@@ -81,33 +84,42 @@ func (pp *PathProcessor) PathEnd2Messages(message string) SequenceCache {
 
 // Path Processors are constructed before ChainProcessors, so reference needs to be added afterwards
 // This can be done inside the ChainProcessor constructor for simplification
-func (pp *PathProcessor) SetChainProcessorIfApplicable(chainID string, chainProcessor ChainProcessor) bool {
-	if pp.pathEnd1.info.ChainID == chainID {
-		pp.pathEnd1.chainProcessor = chainProcessor
+func (pp *PathProcessor) SetChainProviderIfApplicable(chainProvider provider.ChainProvider) bool {
+	if chainProvider == nil {
+		return false
+	}
+	if pp.pathEnd1.info.ChainID == chainProvider.ChainId() {
+		pp.pathEnd1.chainProvider = chainProvider
 		return true
-	} else if pp.pathEnd2.info.ChainID == chainID {
-		pp.pathEnd2.chainProcessor = chainProcessor
+	} else if pp.pathEnd2.info.ChainID == chainProvider.ChainId() {
+		pp.pathEnd2.chainProvider = chainProvider
 		return true
 	}
 	return false
 }
 
+// ProcessBacklogIfReady gives ChainProcessors a way to trigger the path processor process
+// as soon as they are in sync for the first time, even if they do not have new messages.
+func (pp *PathProcessor) ProcessBacklogIfReady() {
+	pp.retryProcess <- struct{}{}
```

Comment:
This will block if the `retryProcess` channel is already at capacity.

I think a slightly better approach is a select:

```go
select {
case pp.retryProcess <- struct{}{}:
  // All good.
default:
  // Log that the channel is saturated;
  // something is wrong if we are retrying this quickly.
  pp.log.Info("Failed to enqueue path processor retry")
}
```

Logging seems appropriate here, as it is likely an issue with the code if the channel is full.

Reply:
done. Called this for the existing retry case also.